cfme/tests/test_db_restore.py

```diff
@@ -0,0 +1,120 @@
+# -*- coding: utf-8 -*-
+
+import fauxfactory
+import pytest
+
+from cfme.configure import configuration as config
+from cfme.cloud import provider as cloud_provider
+from cfme.infrastructure.provider import wait_for_a_provider
+from cfme.infrastructure.virtual_machines import Vm
+from fixtures.pytest_store import store
+from utils.appliance import provision_appliance
+from utils.log import logger
+from utils.providers import setup_a_provider
+from utils import testgen, version
+
+pytest_generate_tests = testgen.generate(testgen.provider_by_type, ['virtualcenter'],
```

Comment:
Now, forgive me for this, but, we need to do one of two things:

1. Replace this line with this,
```python
pytest_generate_tests = testgen.generate(testgen.provider_by_type, ['virtualcenter'],
                                         choose_random=True, scope='module')
```
and remove the definition on lines 20-25

Or and I think the second is my favorite, and I know I mentioned it before and kinda backed down, but I see more robustness in the second method

2. Use ```utils.providers.setup_a_provider``` method. 
```python
setup_a_provider(prov_class=None, prov_type=None, validate=True, check_existing=True)
```
You see this method does a much better job of setting up the provider and provides protection in case there are errors setting up the providers on the appliance. You could call it as a function inside your test on line 77 where you run the ```provider.create()```. ```setup_a_provider``` does a much better job of ensuring that the provider is setup and also gives you repeatability of trying the providers in the same order each time. I think this is a much better idea. It also returns the provider object, so you can still get to the management system.